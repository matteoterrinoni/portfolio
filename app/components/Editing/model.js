export const baseKey = 'editing_';