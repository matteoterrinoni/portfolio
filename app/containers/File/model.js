export const methodTypes = {
	upload : 'upload',
	listView : 'list-view',
	view: 'view'
}

export const nameToKey = (name) => {
	return name.replace(/ /g, '_');
}

export const fileKey = (file) => {
	return nameToKey(file.name);
}