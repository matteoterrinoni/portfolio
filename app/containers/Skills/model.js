import { pickBy, propEq, sortWith, prop, descend } from 'ramda';

export const properties = {
	title: {
		id:'title',
		label: 'Title',
		visible: true
	},
	master: {
		id: 'master',
		label: 'Master',
		visible: true,
		type: 'multipleinput'
	},
	experienced: {
		id: 'experienced',
		label: 'Experienced',
		visible: true,
		type: 'multipleinput'
	},
	novice: {
		id: 'novice',
		label: 'Novice',
		visible: true,
		type: 'multipleinput'
	},
	order: {
		id:'order',
		label: 'Order',
		visible: true
	}
}

export const shallowProperties = {
	master: properties.master,
	experienced: properties.experienced,
	novice: properties.novice
}

export const articleProperties = {
	master: properties.master,
	experienced: properties.experienced,
	novice: properties.novice,
};

const filterVisible = (props) => () => {
	return pickBy(propEq('visible', true), props)
}

export const given = (props)=>{
	return {
		filterVisible: filterVisible(props),
		result: props
	}
}

export const SkillPropsApi = {
	given,
	properties,
	shallowProperties
}

export const visibleProperties = given(properties).filterVisible();

export const sortByOrder = sortWith([descend(prop(properties.order.id))]);

const convert = (skills) => {
	return Object.keys(skills).map(k=>skills[k]);
}

export const skillsToSortedArray = (skills) => {
	return sortByOrder(convert(skills))
}

export const skillStatuses = {
	updated: 'updated',
	toBeUpdated: 'toBeUpdated'
}
