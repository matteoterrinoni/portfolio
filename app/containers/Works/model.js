import { pickBy, propEq, sortWith, prop, descend } from 'ramda';

export const properties = {
	date: {
		id: 'date',
		label: 'Date',
		visible: true
	},
	for: {
		id: 'for',
		label: 'For',
		visible: true
	},
	key: {
		id: 'key',
		label: 'Key',
		visible: false
	},
	position: {
		id: 'position',
		label: 'Position',
		visible: true
	},
	title: {
		id: 'title',
		label: 'Title',
		visible: true
	},
	what: {
		id: 'what',
		label: 'What',
		visible: true
	},
	where: {
		id: 'where',
		label: 'Where',
		visible: true
	},
	html: {
		id: 'html',
		label: 'Html',
		visible: true,
		type: 'code'
	},
	link: {
		id: 'link',
		label: 'Link',
		visible: true
	},
	technologies: {
		id: 'technologies',
		label: 'Technologies',
		visible: true
	},
	order: {
		id:'order',
		label: 'Order',
		visible: true
	},
	images: {
		id:'images',
		label:'Images',
		visible: true,
		type: 'files'
	}
}

export const shallowProperties = {
	for: properties.for,
	what: properties.what,
	technologies: properties.technologies
}

export const articleProperties = {
	position: properties.position,
	date: properties.date,
	for: properties.for,
	technologies: properties.technologies
}

const filterVisible = (props) => () => {
	return pickBy(propEq('visible', true), props)
}

export const given = (props)=>{
	return {
		filterVisible: filterVisible(props),
		result: props
	}
}

export const WorkPropsApi = {
	given,
	properties,
	shallowProperties
}

export const visibleProperties = given(properties).filterVisible();

export const sortByOrder = sortWith([descend(prop(properties.order.id))]);

const convert = (works) => {
	return Object.keys(works).map(k=>works[k]);
}

export const worksToSortedArray = (works) => {
	return sortByOrder(convert(works))
}

export const workStatuses = {
	updated: 'updated',
	toBeUpdated: 'toBeUpdated'
}
