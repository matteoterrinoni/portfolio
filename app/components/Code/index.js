/**
*
* Code
*
*/

import React from 'react';
import { baseKey } from 'components/Editing/model';
import AceEditor from 'react-ace';
import brace from 'brace';

import 'brace/mode/html';
import 'brace/theme/monokai';

import './style.scss';
// import styled from 'styled-components';


class Code extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
	static key = baseKey+'code';
	render() {
		const {label, value, onChange} = this.props;
		return (
			<AceEditor
			mode="html"
			theme="monokai"
			name="code"
			onLoad={null}
			onChange={e=>onChange(e)}
			fontSize={14}
			showPrintMargin={false}
			showGutter={false}
			highlightActiveLine={true}
			value={value}
			setOptions={{
				enableBasicAutocompletion: false,
				enableLiveAutocompletion: false,
				enableSnippets: false,
				showLineNumbers: false,
				tabSize: 2
			}}/>
			);
	}
}

Code.propTypes = {

};

export default Code;
