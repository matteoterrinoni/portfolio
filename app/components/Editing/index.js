import Input from 'components/Input';
import Textarea from 'components/Textarea';
import Code from 'components/Code';
import Files from 'components/Files';

export default [ Input, Textarea, Code, Files];