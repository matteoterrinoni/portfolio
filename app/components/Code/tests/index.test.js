import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { baseKey } from 'components/Editing/model';
configure({ adapter: new Adapter() });

import Code from '../';

describe('<Code />', () => {

	it('should render the proper HTML of Code', () => {
		const wrapper = render(
			<Code value="###" onChange={()=>null}/>
		);
		expect(toJson(wrapper)).toMatchSnapshot()
	});

	it('should have a value for ReactAce attribute', () => {
		const renderedComponent = mount(<Code value="###" onChange={()=>null}/>);
		expect(renderedComponent.find('ReactAce').prop('value')).toBe('###');
	});

	it('should have a static prop baseKey_code', () => {
		const expectedKey = baseKey+'code';
		expect(Code.key).toBe(expectedKey);
	});

});