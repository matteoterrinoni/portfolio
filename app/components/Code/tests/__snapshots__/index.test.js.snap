// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> should render the proper HTML of Code 1`] = `
<div
  id="code"
  style="width:500px;height:500px;"
>
  
</div>
`;
